pages/c/_cluster/index.vue

```diff
@@ -1,27 +1,243 @@
 <script>
+// import { isEmpty } from 'lodash';
 import { get } from '@/utils/object';
-
-import { MANAGEMENT } from '@/config/types';
+import InfoBoxCluster from '@/components/InfoBoxCluster';
+import InfoBox from '@/components/InfoBox';
+import SortableTable from '@/components/SortableTable';
+import {
+  MESSAGE,
+  NAME,
+  NAMESPACE_NAME,
+  REASON,
+  ROLES,
+  STATE,
+  TYPE,
+} from '@/config/table-headers';
+import { DESCRIPTION } from '@/config/labels-annotations';
+import { findAllConstraints } from '@/utils/gatekeeper/util';
+import { MANAGEMENT, EVENT, NODE, METRIC } from '@/config/types';
+import { allHash } from '@/utils/promise';
 
 export default {
+  components: {
+    InfoBox,
+    InfoBoxCluster,
+    SortableTable
+  },
+
+  data() {
+    const constraintHeaders = [
+      NAME,
+      {
+        name:  'Violations',
+        label: 'Violations',
+        value: 'status.totalViolations',
+        sort:  'status.totalViolations',
+        width: 65
+      },
+      STATE,
+    ];
+
+    const reason = { ...REASON, ...{ width: 100 } };
+    const type = { ...TYPE, ...{ width: 100 } };
+    const eventHeaders = [
+      NAMESPACE_NAME,
+      type,
+      reason,
+      {
+        name:  'Object',
+        label: 'Object',
+        value: 'involvedObject.kind',
+        sort:  'involvedObject.kind',
+        width: 100
+      },
+      MESSAGE,
+      {
+        name:      'Last Seen',
+        label:     'Last Seen',
+        value:     'lastTimestamp',
+        sort:      'lastTimestamp',
+        formatter: 'Date',
+        width:     125
+      },
+    ];
+
+    const nodeHeaders = [
+      NAMESPACE_NAME,
+      ROLES,
+      STATE,
+    ];
+
+    return {
+      constraintHeaders,
+      eventHeaders,
+      nodeHeaders
+    };
+  },
+
   async asyncData(ctx) {
     const { route, store } = ctx;
     const id = get(route, 'params.cluster');
 
     const cluster = await store.dispatch('management/find', { type: MANAGEMENT.CLUSTER, id });
 
-    return { cluster };
-  }
+    return {
+      constraints:       [],
+      nodes:             [],
+      events:            [],
+      nodeMetrics:       [],
+      pollingErrorCount: 0,
+      cluster,
+    };
+  },
+
+  async created() {
+    const resourcesHash = await allHash({
+      allNodes:       this.fetchClusterResources(NODE),
+      allEvents:      this.fetchClusterResources(EVENT),
+      rawConstraints: this.fetchConstraints(),
+    });
+
+    this.constraints = resourcesHash.rawConstraints;
+    this.events = resourcesHash.allEvents;
+    this.nodes = resourcesHash.allNodes;
+    this.pollMetrics();
```

Comment:
This should go in `mounted()`, because you don't want it running in SSR (mounted only happens client-side)